src/main/java/fr/openent/swarm/services/impl/DefaultUserService.java

```diff
@@ -0,0 +1,72 @@
+package fr.openent.swarm.services.impl;
+
+import fr.openent.swarm.services.UserService;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.entcore.common.neo4j.Neo4j;
+import org.entcore.common.neo4j.Neo4jResult;
+
+import static fr.openent.swarm.core.constants.Field.CLASSES;
+import static fr.openent.swarm.core.constants.Field.STRUCTURES;
+
+public class DefaultUserService implements UserService {
+    private static final Logger log = LoggerFactory.getLogger(DefaultUserService.class);
+
+    public Future<JsonArray> getUsersByMEF(String userId, JsonArray mefIds){
+        Promise<JsonArray> promise = Promise.promise();
+        this.getUserInfos(userId)
+                .onSuccess(userInfos -> {
+                    String query = "MATCH (u:User) " +
+                            "WHERE u.profiles = [\"Student\"] " +
+                            "AND ANY(id IN {listIdMef} WHERE id = u.module) " +
+                            "AND ANY(structure IN u.structures WHERE structure IN {userStructures}) " +
+                            "AND ANY(classe IN u.classes WHERE classe IN {userClasses}) " +
+                            "OPTIONAL MATCH (s:Structure) WHERE s.externalId IN u.structures " +
+                            "OPTIONAL MATCH (c:Class) WHERE c.externalId IN u.classes " +
+                            "OPTIONAL MATCH (g:Group) WHERE g.externalId IN u.groups " +
+                            "RETURN u.id AS id, " +
+                            "u.firstName AS firstName, " +
+                            "u.lastName AS lastName, " +
+                            "COLLECT(DISTINCT{structureId: s.externalId, name: s.name}) AS structures, " +
+                            "COLLECT(DISTINCT{classId: c.id, name: c.name}) AS classes, " +
+                            "CASE WHEN g IS NULL THEN [] " +
+                            "ELSE COLLECT(DISTINCT {groupId: g.externalId, name: g.name}) END AS groups";
+                    JsonObject params = new JsonObject()
+                            .put("listIdMef", mefIds)
+                            .put("userStructures", userInfos.getJsonArray(STRUCTURES))
+                            .put("userClasses", userInfos.getJsonArray(CLASSES));
```

Comment:
Tu peux utiliser ton fichier constants pour remplacer les magic string ici aussi. Et ajouter "profile" pour le comm précédent
```suggestion
                            .put(LIST_ID_MEF, mefIds)
                            .put(USER_STRUCTURES, userInfos.getJsonArray(STRUCTURES))
                            .put(USER_CLASEES, userInfos.getJsonArray(CLASSES))
                            .put(PROFILE, STUDENT);
```

Reply:
Done